core/iwasm/include/wasm_export.h

```diff
@@ -864,6 +867,37 @@ wasm_runtime_create_exec_env(wasm_module_inst_t module_inst,
 WASM_RUNTIME_API_EXTERN void
 wasm_runtime_destroy_exec_env(wasm_exec_env_t exec_env);
 
+
+typedef bool (*wasm_frame_callback)(void*, wasm_frame_ptr_t);
+
+/**
+ * @brief Iterate over callstack frames and execute callback on it.
+ *
+ * Caution: This is not a thread-safe function. Ensure the exec_env 
+ * is suspended before calling it from another thread.
+ *
+ * Usage: In the callback to read frames fields use APIs 
+ * for wasm_frame_t from wasm_c_api.h
+ *
+ * Note: The function is async-signal-safe if called with verified arguments. 
+ * Meaning it's safe to call it from a signal handler even on a signal interruption
+ * from another thread if next variables hold valid pointers
+ * - exec_env
+ * - exec_env->module_inst
+ * - exec_env->module_inst->module
+ *
+ * Note for devs: please refrain from such modifications inside of this call
```

Comment:
I think it's fine to just keep it in the `.c` file, no need to expose it publicly.

Reply:
Okay, the modifications are done there anyway